### PR TITLE
String word match

### PR DESCRIPTION
**Pull request by Abby3017**
I am adding two programs in this repo.

1. First one is string word match in string folder, where you search for a string in a m*n grid.
2. In dynamic programming folder, I have added program to find maximum product of subarray.

### CODE REVIEW

**Comment by addy1997** [comment]
Hi @Abby3017, thanks for your pr. Could you make the following changes/additions

1. reduce spaces between the braces
2. Implement test files for the same.

**Comment by Abby3017** [comment]
For your first suggestion, can you put the comment on my code review. It will help me to identify which braces you are talking about

File: dynamic programming/max_product_subarray/max_product_subarray.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+)
+
+/*
+Given an array that contains both positive and negative integers,
```
Comment (addy1997) [inline dynamic programming/max_product_subarray/max_product_subarray.go:8]: comments can be written above the package main

**Review by addy1997 — COMMENTED** [review]
Please reduce the spaces between the braces and the comments.

**Comment by Abby3017** [comment]
> Please reduce the spaces between the braces and the comments.

I have reduced the space as suggested by you. If you think any other changes needed, you can comment on code like https://github.com/features/code-review/ 
Please go through Discuss code section in above link

**Pull request merged**